libff/algebra/curves/profile/algebra_groups_read_profile.cpp

```diff
@@ -75,40 +90,647 @@ bool profile_group_read_sequential_uncompressed(const std::string &identifier)
+{
+public:
+    batched_aio_reader(int fd)
+        : _fd(fd)
+        , _active_batch(_batch1_ptrs)
+        , _next_batch(_batch2_ptrs)
+        , _next_batch_size(0)
+    {
+        memset(_batch1, 0, sizeof(_batch1));
+        memset(_batch2, 0, sizeof(_batch2));
+        for (size_t i = 0; i < BATCHSIZE; ++i) {
+            _batch1_ptrs[i] = &_batch1[i];
+            _batch2_ptrs[i] = &_batch2[i];
+        }
+    }
+
+    void enqueue_read_first_batch(
+        size_t offset_bytes, size_t size_bytes, void *dest)
+    {
+        assert(_next_batch_size < BATCHSIZE);
+
+        aiocb *cb = _next_batch[_next_batch_size];
+        cb_init(cb, _fd, offset_bytes, size_bytes, dest);
+
+        // When first batch is full, enqueue it and allow writing to the next
+        // batch.
+
+        ++_next_batch_size;
+        if (_next_batch_size == BATCHSIZE) {
+            enqueue_next_batch();
+
+            std::swap(_active_batch, _next_batch);
+            _next_batch_size = 0;
+        }
+    }
+
+    void enqueue_read(size_t offset_bytes, size_t size_bytes, void *dest)
+    {
+        assert(_next_batch_size < BATCHSIZE);
+
+        aiocb *cb = _next_batch[_next_batch_size];
+        cb_init(cb, _fd, offset_bytes, size_bytes, dest);
+
+        ++_next_batch_size;
+        if (_next_batch_size == BATCHSIZE) {
+            enqueue_next_batch();
+        }
+    }
+
+    void wait_last_read()
+    {
+        // Wait for the _active_batch
+        for (size_t i = 0; i < BATCHSIZE; ++i) {
+            int r = cb_wait(_active_batch[i]);
+            if (0 > r) {
+                throw std::runtime_error("bad read result");
+            }
+            if (0 == r) {
+                std::cout << "+";
+            }
+        }
+
+        // swap pointers so it can be written into
+        std::swap(_active_batch, _next_batch);
+        _next_batch_size = 0;
+    }
+
+protected:
+    void enqueue_next_batch()
+    {
+        int r = lio_listio(LIO_NOWAIT, _next_batch, BATCHSIZE, nullptr);
+        if (0 != r) {
+            throw std::runtime_error("enqueue_batch error");
+        }
+    }
+
+    const int _fd;
+    aiocb _batch1[BATCHSIZE];
+    aiocb _batch2[BATCHSIZE];
+    aiocb *_batch1_ptrs[BATCHSIZE];
+    aiocb *_batch2_ptrs[BATCHSIZE];
+    aiocb **_active_batch; // Next batch to wait for
+    aiocb **_next_batch;   // Batch being filled
+    size_t _next_batch_size;
+};
+
+/// Similar to profile_group_read_random_aio_ordered_uncompressed, but with
+/// more aio requests in-flight at the same time.
+template<typename GroupT>
+void profile_group_read_random_batch_aio_ordered_uncompressed(
+    const std::string &identifier, const size_t interval)
+{
+    const std::string filename = get_filename(identifier);
+
+    // Treat the file as being divided into NUM_ELEMENTS_TO_READ groups, of
+    // size NUM_ELEMENTS_IN_FILE / NUM_ELEMENTS_TO_READ, reading from a ranadom
+    // offset within each group.
+
+    const size_t GROUP_SIZE = interval;
```

Comment:
Why not passing `group_size`  directly as `const size_t` to the function? i.e. why not remove this line and rename the `const size_t interval` param? Is that to keep constants in ALL_CAPS throughout the function code?

Reply:
This should have been changed as part of addressing a comment above.